### ISSUE TITLE
12-configure-pod-networking: network plugin is not ready: cni config uninitialized

### ISSUE THREAD

**Issue by ekovacs** [mmumshad/kubernetes-the-hard-way#119]
on kubernetes 1.18
by following the steps the weave-net daemonset could not start up.

```
vagrant@master-1:~$ k get pods -A                                                                                                                                                  
NAMESPACE     NAME              READY   STATUS             RESTARTS   AGE                                                                                                          
kube-system   weave-net-ltww2   1/2     CrashLoopBackOff   9          32m                                                                                                          
kube-system   weave-net-zzjt9   1/2     CrashLoopBackOff   9          32m 
```

`sudo journalctl -u kubelet.service` on the worker nodes showed lots of errors with:

```
Unable to update cni config: no networks found in /etc/cni/net.d
Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
```

possibly caused by the fact that `/etc/cni/net.d/` had no configs in it at all / nor did the deploying the weave daemonset create it.

i had to 

```
sudo curl -L git.io/weave -o /usr/local/bin/weave
sudo chmod a+x /usr/local/bin/weave
sudo weave setup
```

which created the default `10-weave.conflist` in `/etc/cni/net.d/`

after this, the nodes got ready:
```
 kubectl get nodes
NAME       STATUS   ROLES    AGE     VERSION
worker-1   Ready    <none>   6d1h    v1.18.0
worker-2   Ready    <none>   5d21h   v1.18.0
```

**Comment by realzhangqingwei** [comment]
I have the same problems ,  I cant not solve it

**Comment by realzhangqingwei** [comment]
Failed create pod sandbox: open /run/systemd/resolve/resolv.conf: no such file or directory

**Comment by fireflycons** [comment]
HI @ekovacs @realzhangqingwei 

We have now merged a major change to bring this to v1.24.
Many issues have been addressed as part of the upgrade. 
I realise this is a very old issue, however I was tasked with closing off all old issues following the upgrade.
Please try it now and feel free to raise further issues.

Thanks.
